src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1645,6 +1645,36 @@ public bool Equals(StringBuilder sb)
             }
         }
 
+        /// <summary>
+        /// Determines if the contents of this builder are equal to the contents of ReadOnlySpan<char>.
+        /// </summary>
+        /// <param name="value">The ReadOnlySpan<char>.</param>
+        public bool Equals(ReadOnlySpan<char> value)
+        {
+            if (value.Length != Length)
+                return false;
+
+            StringBuilder thisChunk = this;
+            int thisChunkIndex = thisChunk.m_ChunkLength;
+
+            for (int i = Length - 1; i >= 0; i--)
+            {
+                --thisChunkIndex;
+                while (thisChunkIndex < 0)
```

Comment:
Why this isn't this just `if (thisChunkIndex < 0)`

Reply:
I think if should work and I have changed it to IF. But I am still not able to find out why While was used in the first place.